src/transform/plugins/changelog/collect.ts

```diff
@@ -28,33 +28,38 @@ type Options = Pick<MarkdownItPluginOpts, 'path' | 'log'> & {
 const collect = (input: string, {path: filepath, log, changelogs, extractChangelogs}: Options) => {
     let result = input;
     let lastPos = 0;
-    const rawChanges = [];
+    const rawChangelogs = [];
 
     // eslint-disable-next-line no-constant-condition
     while (true) {
-        const pos = result.indexOf(BLOCK_START, lastPos);
-        lastPos = pos;
-        if (pos === -1) {
+        const startPos = result.indexOf(BLOCK_START, lastPos);
+        lastPos = startPos;
+        if (startPos === -1) {
             break;
         }
-        const endPos = result.indexOf(BLOCK_END, pos + BLOCK_START.length);
-        if (endPos === -1) {
+
+        const endBlockPos = result.indexOf(BLOCK_END, startPos + BLOCK_START.length);
+        if (endBlockPos === -1) {
             log.error(`Changelog block must be closed${filepath ? ` in ${bold(filepath)}` : ''}`);
             break;
         }
+        let endPos = endBlockPos + BLOCK_END.length;
+        if (result[endPos + 1] === '\n') {
```

Comment:
I suppose you should add a test about it

Reply:
Done! Thx!